addons/delivery/models/sale_order.py

```diff
@@ -73,4 +70,29 @@ def delivery_set(self):
 
             else:
                 # Classic grid-based carriers
-                super(SaleOrder, self).delivery_set()
+                carrier = order.carrier_id.carrier_get(order.partner_shipping_id)
```

Comment:
it seem duplicate code. 

I think, just need to get  price base on carrier delivery type.